src/main/java/org/isf/accounting/gui/PatientBillEdit.java

```diff
@@ -814,71 +922,50 @@ private JScrollPane getJScrollPaneTotal() {
 
 	private JTable getJTableBigTotal() {
 		if (jTableBigTotal == null) {
-			jTableBigTotal = new JTable();
-			jTableBigTotal.setModel(new DefaultTableModel(new Object[][] {
-					{
-							"<html><b>" + MessageBundle.getMessage("angal.newbill.topay.txt") + "</b></html>",
-							currencyCod,
-							bigTotal}
-			}, new String[] {"","", ""}) {
-				private static final long serialVersionUID = 1L;
-				Class<?>[] types = new Class<?>[] { JLabel.class, JLabel.class, Double.class, };
-
-				@Override
-				public Class<?> getColumnClass(int columnIndex) {
-					return types[columnIndex];
-				}
-
-				@Override
-				public boolean isCellEditable(int row, int column) {
-					return false;
-				}
-			});
-			jTableBigTotal.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
-			jTableBigTotal.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
-			jTableBigTotal.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
-			jTableBigTotal.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
-			jTableBigTotal.setMaximumSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
-			jTableBigTotal.setMinimumSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
-			jTableBigTotal.setPreferredSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
+			jTableBigTotal = new JTable(new JTableBigTotalModel());
+			sizeJTableBigTotal();
 		}
 		return jTableBigTotal;
 	}
 
+	private void sizeJTableBigTotal() {
+		jTableBigTotal.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
+		jTableBigTotal.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
+		jTableBigTotal.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
+		jTableBigTotal.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
+		jTableBigTotal.setMaximumSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
```

Comment:
wouldn't it be better to create the Dimension object once and refer to it 3 times?